src/App.js

```diff
@@ -18,14 +18,62 @@ import { allMessages } from './locales/languages';
 import languages from './components/AppStepper/VersionStep/Variables/Languages';
 import availableFeatures from './components/AppStepper/FeaturesStep/AvailableFeatures';
 
-let currentLocale = navigator.language.split(/[-_]/)[0];
-console.log(`Detected browser language: ${currentLocale}`);
+const browserLanguages = navigator.languages ||
+  navigator.language ||
+  navigator.browserLanguage ||
+  navigator.userLanguage;
+console.log(`Detected browser languages: [${browserLanguages.join(', ')}]`);
```

Comment:
If user comes from a country not supported by the GUI he/she got en_GB now. If he/she is not familiar with english and his/her browser is setted with other supported language with preferred order, then he/she got the preferred one. `userLanguage` and `browserLanguage ` for IE.

Reply:
IE should not be considered at all. MS is dropping support for it:
https://blogs.windows.com/windowsexperience/2021/05/19/the-future-of-internet-explorer-on-windows-10-is-in-microsoft-edge/
And there are bazillion other reasons why IE should not be used :) 

here is my `navigator.languages`:
![image](https://user-images.githubusercontent.com/15658235/119805270-65d04880-bee1-11eb-9238-649f59846f21.png)
and the order is changing depending on what I'm currently doing :)

There is no need to complicate TC source code to include scenario when someones browser has more than one language on the list. The `navigator.language` is perfectly fine to dermine what TC or Tasmota UI language should be set by default and if it is not on the list of supported languages... well it will be english by default.
The whole documentation, templates, forum, discord channel and probably many other things related to Tasmota are ONLY available in English.

Reply:
With my english I can't argue well, and of course it's yout subject to merge or reject.
Not much, I just do my best to help the userbase make firmware binaries with less interact :)

Reply:
You have PRed a lot of good changes, and I really appreciate your contribution (and I think not only me). This time I just don't see good reason to implement proposed changes ;)

Reply:
I am happy to do and will continue to do so.
In my case it still work, so who cares :)
Seriously, that's how it works. PR can be closed.